### ISSUE TITLE
[RFC] Add a slot `ActionBar` for all pickers

### ISSUE THREAD

**Issue by flaviendelangle** [mui/mui-x#4655]
## Goals

- Unify how various actions are enabled / disabled
- Unify available actions across all components (by removing props drilling which was the source of bugs because we forgot to drill some props on some components)
- Simplify typing by removing props on the top-level components

## Current behavior

The actions of the mobile pickers are handled by `PickersModalDialog` as follow:

- show "Clear" button if `props.clearable = true`
- show "Today" button if `props.showTodayButton = true`
- show "Cancel" button if `!!props.cancelText`
- show "OK" button if `!!props.okText`

The actions of the desktop pickers are handled by `PickersPopper` as follow:

- show "Clear" button if `props.clearable = true`

## Proposed behaviors

Add a new `PickerActionBar` common to all pickers, this component would be customized through the `props.components` and `props.componentsProps`. It would be added by default to the mobile pickers but could be added very easily to the desktop and static pickers.

### Removed props

- `props.showTodayButton`
- `props.clearable`

### Proposal A 

```tsx
import { PickerActionBar, PickerApproveButton, PickerCancelButton, PickerTodayButton, PickerClearButton } from '@mui/x-date-pickers/PickerActionBar'

<DesktopDatePicker
  // Need to be activated, by default it is not enabled on desktop
  components={{ ActionBar: PickerActionBar  }}

  // By default the actions would be `[<PickerApproveButon />, <PickerCancelButton />]`
  componentsProps={{ actionBar: { 
    actions: [
      <PickerTodayButton />, 
      <PickerClearButton />, 
      <PickerApproveButton />,
    ] }}}
/>
```

##### Pro

- Easy for users to create custom actions 
- Easy for users to change the order of the actions

##### Cons

- Verbose
- A lot of very small components to export

### Proposal B 

```tsx
<DesktopDatePicker
  // Need to be activated, by default it is not enabled on desktop
  components={{ ActionBar: PickerActionBar  }}
  componentsProps={{
    actionBar: { showTodayButton: true, showCancelButton: false }
  }}
/>
```

##### Pro

- Concise
- Easy to understand

##### Cons

- Harder to customize (but users could provide a custom `ActionBar` and re-implement the buttons which are very simple themselved
- Can't easily change the action order

### Proposal C

```tsx
<DesktopDatePicker
  // Need to be activated, by default it is not enabled on desktop
  components={{ ActionBar: PickerActionBar  }}
  componentsProps={{
    actionBar: { actions: ['today', 'clear', 'approve'] },
  }}
/>
```

##### Pro

- Concise
- Easy to understand
- Easy for users to change the order of the actions

##### Cons

- Harder to customize (but users could provide a custom `ActionBar` and re-implement the buttons which are very simple themselved

### Related issues

- #4505
- #4596

**Comment by flaviendelangle** [comment]
@joserodolfofreitas @alexfauquette

**Comment by joserodolfofreitas** [comment]
I like **Proposal C** the most.
It seems the cleanest and with enough quick customization. The string input for actions might not provide the best dx with auto-complete, but should we consider an enum for it? Or would it be too verbose?

Btw, what would the default actions be?

**Comment by flaviendelangle** [comment]
> The string input for actions might not provide the best dx with auto-complete, but should we consider an enum for it? 

You can type the string

```ts
type PickerActionBarAction = 'clear' | 'cancel' | 'accept' | 'today'

type PickerActionBarProps {
  actions?: PickerActionBarAction[]
}
```

**Comment by flaviendelangle** [comment]
Both B and C are fine for me, so if you both prefer C let's do C !

I propose to keep the differences between the pickers as limited as possible.
To always have the same default actions ([`cancel`, `accept`] I think) and to just enable or not the action bar by default (activated on mobile, not on desktop and static I think)

**Comment by flaviendelangle** [comment]
Closes by #4778

**Comment by flaviendelangle** [comment]
My bad, it still does not work on static pickers

**Comment by flaviendelangle** [comment]
Fixed by #5015
